lib/core/src/Cardano/Wallet.hs

```diff
@@ -288,13 +291,27 @@ data WalletLayer s t = WalletLayer
     , listTransactions
         :: DefineTx t
         => WalletId
-        -> ExceptT ErrNoSuchWallet IO [(Tx t, TxMeta)]
+        -> ExceptT ErrNoSuchWallet IO [TransactionInfo]
         -- ^ List all transactions and metadata from history for a given wallet.
         --
         -- The result is sorted on 'slotId' in descending order. The most recent
         -- transaction comes first.
     }
 
+-- | Full information about a transaction, suitable for presentation to the
+-- user.
+data TransactionInfo = TransactionInfo
+    { txId :: Hash "Tx"
+    -- ^ Transaction ID of this transaction
+    , inputs :: [(TxIn, Maybe TxOut)]
+    -- ^ Transaction inputs and (maybe) corresponding outputs of the
+    -- source. Source information can only be provided for outgoing payments.
+    , outputs :: [TxOut]
+    -- ^ Payment destination.
+    , infoMeta :: TxMeta
+    -- ^ Other information calculated from the transaction.
+    } deriving (Show, Eq, Ord)
```

Comment:
:+1: 

I'd rather have this type lives in `Cardano.Wallet.Primitive.Types` though, close to the other for an easier maintainability & consistency,